pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -64,6 +72,7 @@ type Provisioner struct {
 	RKESystemImagesLister v3.RkeK8sSystemImageLister
 	SecretLister          corev1.SecretLister
 	Secrets               corev1.SecretInterface
+	clusterManager        wrangler.MultiClusterManager
```

Comment:
I don't think that it's a good idea to pass in the entire MCM to the provisioner. Rather, I would recommend that you just pass in the `clustermanager.Manager`.  

To do this, you will need to:
- Modify the `Register` function to accept a `manager clustermanager.Manager` as a third parameter.
- Modify the [calling function](https://github.com/rancher/rancher/blob/1913156a9f32ded561ab100ff0ce7f9135ad8e85/pkg/controllers/management/controller.go#L49) to pass in the manger which it already has.

Reply:
I cannot use `clustermanager.Manager` -- this causes a cyclical dependency in this package.

Reply:
I see - sorry I did not see that before. You can get around this by implementing an interface that supports the method that you from the `clustermanager.Manager`. I was able to get the following to compile on my local:

```go
type UserContextGetter interface {
	UserContextNoControllers(clusterName string) (*config.UserContext, error)
}

func Register(ctx context.Context, management *config.ManagementContext, manager UserContextGetter) {
```

This also follows the recommend "return structs accept interfaces" paradigm that I think is encouraged for go.

Reply:
Updated.

Reply:
Resolving, thanks for making the change.